smart_contract/executor/derive/src/lib.rs

```diff
@@ -195,3 +199,155 @@ pub fn derive_ref_into_account_owner(input: TokenStream) -> TokenStream {
 pub fn derive_ref_into_domain_owner(input: TokenStream) -> TokenStream {
     conversion::impl_derive_ref_into_domain_owner(input)
 }
+
+/// Implements the `iroha_validator::Validate` trait for the given `Executor` struct. As
+/// this trait has a `iroha_validator::prelude::Visit`, and the latter has an
+/// `iroha_validator::iroha_data_model::evaluate::ExpressionEvaluator`
+/// bound, at least these two should be implemented as well.
+///
+/// Emits a compile error if the struct didn't have all the expected fields with corresponding
+/// types, i.e. `verdict`: `iroha_validator::prelude::Result`, `block_height`: `u64` and
+/// `host`: `iroha_validator::iroha_wasm::Host`, though technically only `verdict` and
+/// `block_height` are needed. The types can be unqualified, but not aliased.
+#[manyhow]
+#[proc_macro_derive(Validate)]
+pub fn derive_validate(input: TokenStream2) -> TokenStream2 {
+    let mut emitter = Emitter::new();
+
+    let Some(input) = emitter.handle(syn2::parse2(input)) else {
+        return emitter.finish_token_stream();
+    };
+
+    let result = default::impl_derive_validate(&mut emitter, &input);
+
+    emitter.finish_token_stream_with(result)
+}
+
+/// Implemets the `iroha_validator::prelude::Visit` trait on a given `Executor` struct.
+/// Users can supply custom overrides for any of the visit functions as freestanding functions
+/// in the same module or omit some of the visit functions completely via the `#[omit(visit(...))]`
+/// attribute by supplying corresponding visit function names inside of it, otherwise a default
+/// implementation from `iroha_validator::default` module is used.
```

Comment:
You write ` ... or omit ...` but in fact looks like users *have* to use `omit` to provide their own functions, no?

Reply:
Indeed, good catch in wording.

Reply:
I don't understand how they may not need them if this functions are the part of the `Visit` trait and they will be implemented anyway

Reply:
But as @mversic mentioned above, the default `visit_..` fns just visit the nodes without additional behavior, and the `Visit` impl just brings them into immediate scope. It is true that we're not actually omitting things listed in the attribute. So thinking about it again, maybe it makes more sense to call this `replace` instead of `omit` as that's what's actually being done.

Reply:
Yeah, I'd like `replace` more. Or maybe something like `custom`